modules/uavcan_debug/uavcan_debug.c

```diff
@@ -0,0 +1,47 @@
+#include "uavcan_debug.h"
+#include <modules/uavcan/uavcan.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <string.h>
+#include <chprintf.h>
+#include <memstreams.h>
+void uavcan_send_debug_msg(uint8_t debug_level, char* source, const char *fmt, ...)
+{
+    struct uavcan_protocol_debug_LogMessage_s log_msg;
+
+    va_list ap;
+    MemoryStream ms;
+    BaseSequentialStream *chp;
+
+    /* Memory stream object to be used as a string writer, reserving one
+     byte for the final zero.*/
+    msObjectInit(&ms, (uint8_t *)log_msg.text, sizeof(log_msg.text), 0);
+
+    /* Performing the print operation using the common code.*/
+    chp = (BaseSequentialStream *)(void *)&ms;
+    va_start(ap, fmt);
+    log_msg.text_len = chvprintf(chp, fmt, ap);
+    va_end(ap);
+
+    /* Terminate with a zero, unless size==0.*/
+    if (ms.eos < sizeof(log_msg.text))
+      log_msg.text[ms.eos] = 0;
+
+
+    log_msg.source_len = strlen(source);
```

Comment:
This needs to be protected from overflow with strnlen(source, sizeof(log_msg.source))